### ISSUE TITLE
[REVIEW]: portalcasting: Supporting automated forecasting of rodent populations

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#3220]
**Submitting author:** @gmyenni (<a href="http://orcid.org/0000-0001-6969-1848">Glenda Yenni</a>)
**Repository:** <a href="http://github.com/weecology/portalcasting" target ="_blank">http://github.com/weecology/portalcasting</a>
**Version:** v0.20.0
**Editor:** @marcosvital
**Reviewer:** @ViniciusBRodrigues , @FlukeAndFeather 
**Archive:** Pending

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

## Status

[![status](https://joss.theoj.org/papers/b5f7d943424452cb698c44f7a233fd32/status.svg)](https://joss.theoj.org/papers/b5f7d943424452cb698c44f7a233fd32)

Status badge code:

```
HTML: <a href="https://joss.theoj.org/papers/b5f7d943424452cb698c44f7a233fd32"><img src="https://joss.theoj.org/papers/b5f7d943424452cb698c44f7a233fd32/status.svg"></a>
Markdown: [![status](https://joss.theoj.org/papers/b5f7d943424452cb698c44f7a233fd32/status.svg)](https://joss.theoj.org/papers/b5f7d943424452cb698c44f7a233fd32)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="http://github.com/weecology/portalcasting" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) by leaving comments in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@ViniciusBRodrigues  & @FlukeAndFeather , please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html. Any questions/concerns please let @marcosvital know.

✨ **Please start on your review when you are able, and be sure to complete your review in the next six weeks, at the very latest** ✨ 

### Conflict of interest

- [ ] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [ ] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [ ] **Repository:** Is the source code for this software available at the <a target="_blank" href="http://github.com/weecology/portalcasting">repository url</a>?
- [ ] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [ ] **Contribution and authorship:** Has the submitting author (@gmyenni) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [ ] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [ ] **Installation:** Does installation proceed as outlined in the documentation?
- [ ] **Functionality:** Have the functional claims of the software been confirmed?
- [ ] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [ ] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [ ] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [ ] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [ ] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [ ] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [ ] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [ ] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [ ] **A statement of need:** Does the paper have a section titled 'Statement of Need' that clearly states what problems the software is designed to solve and who the target audience is?
- [ ] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [ ] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [ ] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

### Conflict of interest

- [ ] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [ ] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [ ] **Repository:** Is the source code for this software available at the <a target="_blank" href="http://github.com/weecology/portalcasting">repository url</a>?
- [ ] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [ ] **Contribution and authorship:** Has the submitting author (@gmyenni) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [ ] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [ ] **Installation:** Does installation proceed as outlined in the documentation?
- [ ] **Functionality:** Have the functional claims of the software been confirmed?
- [ ] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [ ] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [ ] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [ ] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [ ] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [ ] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [ ] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [ ] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [ ] **A statement of need:** Does the paper have a section titled 'Statement of Need' that clearly states what problems the software is designed to solve and who the target audience is?
- [ ] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [ ] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [ ] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @ViniciusBRodrigues , @FlukeAndFeather  it looks like you're currently assigned to review this paper :tada:.

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

For example, to regenerate the paper pdf after making changes in the paper's md or bib files, type:

```
@whedon generate pdf
```

**Comment by whedon** [comment]
```
Software report (experimental):

github.com/AlDanial/cloc v 1.88  T=0.12 s (573.3 files/s, 120038.7 lines/s)
-------------------------------------------------------------------------------
Language                     files          blank        comment           code
-------------------------------------------------------------------------------
R                               50           1238           4862           6188
Markdown                         7            232              0            700
YAML                             6             28              2            426
Rmd                              5            287            459            247
TeX                              1              8              0            114
JSON                             1              0              0             60
Dockerfile                       1              5              3              6
-------------------------------------------------------------------------------
SUM:                            71           1798           5326           7741
-------------------------------------------------------------------------------


Statistical information for the repository '3d79fca4c32cf2859425f6db' was
gathered on 2021/04/27.
The following historical commit information, by author, was found:

Author                     Commits    Insertions      Deletions    % of changes
DAPPERstats                      2             6              6          100.00

Below are the number of rows from each author that have survived and are still
intact in the current revision:

Author                     Rows      Stability          Age       % in comments
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03220/joss.03220/10.21105.joss.03220.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03220/joss.03220/10.21105.joss.03220.pdf) :page_facing_up: :point_left:

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.21105/joss.01098 is OK
- 10.1890/08-1222.1 is OK
- 10.1890/15-2115.1 is OK
- 10.1111/2041-210X.13104 is OK
- 10.1101/332783 is OK
- 10.5281/zenodo.1475410 is OK

MISSING DOIs

- 10.1126/science.196.4292.880 may be a valid DOI for title: Competition between seed-eating rodents and ants in desert ecosystems

INVALID DOIs

- None
```

**Comment by marcosvital** [comment]
Dear @gmyenni: your manuscript will be reviewed in this issue, and you can reply any comments and suggestions that the reviewers might address right here.

**Comment by marcosvital** [comment]
@ViniciusBRodrigues and @FlukeAndFeather: thank you again for for accepting review this submission for JOSS.

Even if you are not starting the review right now, please accept the invite, as it has an expiration date (there is a link under Reviewer instructions & questions and you should also get an email notification). Furthermore, please check the instructions and checklists above, and let me know if you need any assistance.

You can also tag  @gmyenni if you need to ask specific questions about the submission or to address any changes that might be necessary in the submitted paper or in the repository.

**Comment by gmyenni** [comment]
Thanks for all your help @marcosvital !

**Comment by ViniciusBRodrigues** [comment]
Hi @marcosvital I thought that I have accepted the invitation here, but when I clicked in the link I received: "Sorry, we couldn't find that repository invitation. It is possible that the invitation was revoked or that you are not logged into the invited account."

Could you check it please?

**Comment by whedon** [comment]
I'm sorry human, I don't understand that. You can see what commands I support by typing:

```
@whedon commands
```

**Comment by danielskatz** [comment]
@whedon re-invite @ViniciusBRodrigues as reviewer

**Comment by whedon** [comment]
OK, the reviewer has been re-invited.

@viniciusbrodrigues please accept the invite by clicking this link: https://github.com/openjournals/joss-reviews/invitations

**Comment by ViniciusBRodrigues** [comment]
@gmyenni can you improve the **Community guidelines** in your GitHub page (and also in https://weecology.github.io/portalcasting/)? There are not *clear guidelines* for that there. For someone which doesn't use GitHub it could be trickier.

**Comment by FlukeAndFeather** [comment]
@gmyenni Two things I want to bring to your attention. The first is pretty minor: a few errors in the manuscript's bibliography. Brown (1998) is missing editors and a few references have incorrect capitalization ("portal" when referring to Portal, AZ and "Portalr" when referring to the "portalr" pkg).

The second issue is: I'm running into errors trying to walk through the Getting Started vignette. After installing portalcasting from GitHub, I encounter the following.

```
library(portalcasting)
main <- "portalcast_directory"
setup_dir(main = main)
portalcast(main = main, models = c("ESSS", "AutoArima", "NaiveArima"))
plot_cast_ts(main = main)
# Error: cast not available for requested species
traceback()
# 2: stop("cast not available for requested species", call. = FALSE)
# 1: plot_cast_ts(main = main)
```

Is the Getting Started vignette out of date? Or is it an issue with my computational environment? If running a Docker image is essential, then (given the target audience) I think the tutorial on spinning up a container is required documentation.

**Comment by marcosvital** [comment]
Hi, @gmyenni. I'm passing by just to check if everything is okay with you, and if you are still able to work on this submission, since there is some time since your last activity in this issue. Let us know if you need any help or need more time to work on this, ok?

**Comment by juniperlsimonis** [comment]
> 
> 
> Hi, @gmyenni. I'm passing by just to check if everything is okay with you, and if you are still able to work on this submission, since there is some time since your last activity in this issue. Let us know if you need any help or need more time to work on this, ok?

Hi @marcosvital! Thanks for checking in on us. I'm the lead on the package updates and experienced some significant personal setbacks recently that have slowed my abilities to physically type, and I've been playing catch up. 
We're still working on this, so would appreciate more time. 
Thanks so much!

**Comment by marcosvital** [comment]
Hi, @juniperlsimonis, I hope everything gets better for you soon. Thank you for letting us know, and take your time.

**Comment by juniperlsimonis** [comment]
Thank you @marcosvital

**Comment by marcosvital** [comment]
Dear @gmyenni and @juniperlsimonis. I hope that everything is fine with both of you.
Please let us know when you are ready to work on this submission again, ok? Also, let me know if you need any assistance from me.

**Comment by gmyenni** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03220/joss.03220/10.21105.joss.03220.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03220/joss.03220/10.21105.joss.03220.pdf) :page_facing_up: :point_left:

**Comment by gmyenni** [comment]
Hi @marcosvital, thanks for checking in. Unfortunately, we are going to need a bit more time before we can get back to this. I would love to get this off everyone's plates, so I will be sure it is ASAP. Many apologies for the delays and thanks so much for your understanding.

**Comment by davan690** [comment]
Hi, still needing a review for this?

**Comment by marcosvital** [comment]
Hi, @gmyenni and @juniperlsimonis!
Please let us know if you think you can work on this submission again, ok?

**Comment by gmyenni** [comment]
Yep, we are finally ready to start this up again! Terribly sorry for the huge delay. Thank you so much for your understanding.
We have updated the documentation, and we've updated and tested the vignettes. @FlukeAndFeather hopefully it works for you now?

**Comment by marcosvital** [comment]
Thank you, @gmyenni and @juniperlsimonis.

@FlukeAndFeather and @ViniciusBRodrigues, please let us know if you can continue to review this submission.

**Comment by ViniciusBRodrigues** [comment]
> Thank you, @gmyenni and @juniperlsimonis.
> 
> @FlukeAndFeather and @ViniciusBRodrigues, please let us know if you can continue to review this submission.

Sure!

**Comment by marcosvital** [comment]
Dear @ViniciusBRodrigues and @FlukeAndFeather, any news on the reviewing process?

**Comment by ViniciusBRodrigues** [comment]
> Dear @ViniciusBRodrigues and @FlukeAndFeather, any news on the reviewing process?

Sorry for the delay. Now the gaps was filled. Many thanks!

**Comment by marcosvital** [comment]
> > Dear @ViniciusBRodrigues and @FlukeAndFeather, any news on the reviewing process?
> 
> Sorry for the delay. Now the gaps was filled. Many thanks!
Thank you, @ViniciusBRodrigues! 

I see there is still one box unchecked. Is this something the authors should address, or did you miss this one?

**Comment by marcosvital** [comment]
@editorialbot check references

**Comment by marcosvital** [comment]
@editorialbot generate pdf

**Comment by marcosvital** [comment]
Dear @gmyenni, I think we are almost done here.

Please take a careful look at the new pdf version of the manuscript above. Let me know if everything is fine or if you need to make any final corrections.

**Comment by gmyenni** [comment]
That looks great. Thank you so much @marcosvital, @ViniciusBRodrigues, @FlukeAndFeather for your help and extreme patience.

**Comment by marcosvital** [comment]
@editorialbot set v0.36.0 as version

**Comment by marcosvital** [comment]
@editorialbot set 10.5281/zenodo.3332973 as archive

**Comment by marcosvital** [comment]
@editorialbot recommend-accept

**Comment by arfon** [comment]
@gmyenni – could you please merge this PR? https://github.com/weecology/portalcasting/pull/276

**Comment by arfon** [comment]
@editorialbot accept

**Comment by arfon** [comment]
@ViniciusBRodrigues & @FlukeAndFeather – many thanks for your reviews here and to @marcosvital for editing this submission! JOSS relies upon the volunteer effort of people like you and we simply wouldn't be able to do this without you ✨ 

@gmyenni – your paper is now accepted and published in JOSS :zap::rocket::boom:

**Comment by FlukeAndFeather** [comment]
@marcosvital @juniperlsimonis @gmyenni My apologies for not responding after you resumed work on this submission! I think I silenced notifications for this thread and forgot to check back in. Congratulations on your paper acceptance!
